src/sqlfluff/rules/structure/ST09.py

```diff
@@ -0,0 +1,284 @@
+            first_column_reference = subcondition[0]
+            second_column_reference = subcondition[2]
+            raw_comparison_operators = comparison_operator.get_children(
+                "raw_comparison_operator"
+            )
+
+            first_table = first_column_reference.get_child("naked_identifier").raw_upper
+            second_table = second_column_reference.get_child(
+                "naked_identifier"
+            ).raw_upper
+
+            # if we swap the two column references around the comparison operator
+            # we might have to replace the comparison operator with a different one
+            raw_comparison_operator_opposites = {"<": ">", ">": "<"}
+
+            # there seem to be edge cases where either the first table or the second
+            # table is not in table_aliases, in which case we cannot provide any fix
+            if first_table not in table_aliases or second_table not in table_aliases:
+                continue
+
+            if (
+                table_aliases.index(first_table) > table_aliases.index(second_table)
+                and self.preferred_first_table_in_join_clause == "earlier"
+            ) or (
+                table_aliases.index(first_table) < table_aliases.index(second_table)
+                and self.preferred_first_table_in_join_clause == "later"
+            ):
+                fixes = (
+                    fixes
+                    + [
+                        LintFix.replace(
+                            first_column_reference,
+                            [second_column_reference],
+                        )
+                    ]
+                    + [
+                        LintFix.replace(
+                            second_column_reference,
+                            [first_column_reference],
+                        )
+                    ]
+                    + (
+                        [
+                            LintFix.replace(
+                                raw_comparison_operators[0],
+                                [
+                                    SymbolSegment(
+                                        raw=raw_comparison_operator_opposites[
+                                            raw_comparison_operators[0].raw
+                                        ],
+                                        type="raw_comparison_operator",
+                                    )
+                                ],
+                            )
+                        ]
+                        if raw_comparison_operators[0].raw
+                        in raw_comparison_operator_opposites
+                        and [r.raw for r in raw_comparison_operators] != ["<", ">"]
+                        else []
+                    )
+                )
+
+        # STEP 5.a.
+        if fixes == []:
+            return None
+
+        # STEP 5.b.
+        else:
+            return LintResult(anchor=context.segment, fixes=fixes)
```

Comment:
One last suggestion here. The current description shown in the test suite is `Joins should list the table referenced earlier/later first`. By setting the `description` argument here you can reflect the current config:

```suggestion
            return LintResult(
                anchor=context.segment,
                fixes=fixes,
                description=f"Joins should list the table referenced {self.preferred_first_table_in_join_clause} first"
            )
```